domain_config_ui/domain_config_ui.services.yml

```diff
@@ -0,0 +1,23 @@
+services:
+  domain_config_ui.negotiator:
+    class: Drupal\domain_config_ui\DomainConfigUINegotiator
+    tags:
+      - { name: persist }
+    arguments: ['@request_stack', '@module_handler', '@domain.loader', '@config.factory']
+    calls:
+      - [setDomainConfigOverrider, ['@domain_config.overrider']]
```

Comment:
Passing the domain overrider to the UI negotiator so that the domain can be switched on AJAX requests